pkg/client/cache/listers.go

```diff
@@ -193,7 +220,13 @@ func (s *StoreToReplicationControllerLister) GetPodControllers(pod *api.Pod) (co
 		return
 	}
 
-	for _, m := range s.Store.List() {
+	key := &api.ReplicationController{ObjectMeta: api.ObjectMeta{Namespace: pod.Namespace}}
+	items, err := s.Indexer.Index(NamespaceIndex, key)
+	if err != nil {
+		return
+	}
+
+	for _, m := range items {
 		rc = *m.(*api.ReplicationController)
 		if rc.Namespace != pod.Namespace {
```

Comment:
Checking the namespace here is no longer needed (because we know that all selected RCs are from this namespace).